File: c#/leet_code/1544.cs
public class Solution {
    public string MakeGood(string s) {
        Stack<char> stack = new Stack<char>();
        foreach (char c in s) {
            if (stack.Count > 0 && Math.Abs(stack.Peek() - c) == 32) {
                stack.Pop();
            } else {
                stack.Push(c);
            }
        }
        char[] result = stack.ToArray();
        Array.Reverse(result);
        return new string(result);
    }
}

File: php/leet_code/1544.php
<?php

class Solution
{

    /**
     * @param String $s
     * @return String
     */
    public function makeGood($s)
    {
        $stack = [];
        for ($i = 0, $iMax = strlen($s); $i < $iMax; $i++) {
            $c = $s[$i];
            if (empty($stack)) {
                $stack[] = $c;
            } else {
                $top = end($stack);
                if (abs(ord($top) - ord($c)) === 32) {
                    array_pop($stack);
                } else {
                    $stack[] = $c;
                }
            }
        }
        return implode('', $stack);
    }
}
